### PR TITLE
fix(natives/five): weather pointer offset

### LINKED ISSUE #2711: Native issue with Weather?
### What happened?

image

Entire server crashed when we turned on "Xmas Snow" setting in vMenu. Have to say its the first time we've ever seen the weather cause crashes.

Has this native been changed recently?

tx: v7.2.2 | fx: b9424
vMenu

Oh?
image

### Expected result

To be able to load "Xmas Snow" weather setting.

### Reproduction steps

1. Set Xmas Snow weather in vMenu v3.6.3

### Importancy

Slight inconvenience

### Area(s)

FiveM, FXServer, OneSync, Natives

### Specific version(s)

FiveM Server Artifacts 9424

### Additional information

_No response_

### LINKED ISSUE #2736: ForceSnowPass native crashes the client
### What happened?

I've tried using the native : `ForceSnowPass(true)` on the client and my game crashed : 
image

### Expected result

Don't crash

### Reproduction steps

1. `ensure runcode`
2. `ForceSnowPass(true)`
3. Crash

### Importancy

Crash

### Area(s)

FiveM

### Specific version(s)

Latest client and artifacts

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by DaniGP17**
### Goal of this PR

Fix crashes with native `FORCE_SNOW_PASS` due to a new weather addition in build 3258.

### How is this PR achieving the goal

Use a specific offset to get the current weather with build 3095 onwards, due to the addition of a new weather type called `SNOW_HALLOWEEN`.

### This PR applies to the following area(s)

FiveM

### Successfully tested on

**Game builds:** 1604, 2060, 2189, 2372, 2545, 2612, 2699, 2802, 2944, 3095, 3258

**Platforms:** Windows

### Checklist

- [x] Code compiles and has been tested successfully.
- [x] Code explains itself well and/or is documented.
- [x] My commit message explains what the changes do and what they are for.
- [x] No extra compilation warnings are added by these changes.

### Fixes issues

fixes #2736, fixes #2711

### CODE REVIEW

File: code/components/extra-natives-five/src/WeatherNatives.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ static HookFunction hookFunction([]()
 		hook::set_call(&getWeatherByName, location + 23);
 		hook::call(location + 23, getWeatherByNameHook);
 
-		g_currentWeather = hook::get_address<int*>(location + 30);
+		g_currentWeather = hook::get_address<int*>(location + (xbr::IsGameBuildOrGreater<3095>() ? 51 : 30));
```
Comment (FabianTerhorst) [inline code/components/extra-natives-five/src/WeatherNatives.cpp:72]: Are you sure the offset changed in 3095 and not in 3258? For me the old offset works fine in 3095.

**Review by FabianTerhorst — CHANGES_REQUESTED** [review]

File: code/components/extra-natives-five/src/WeatherNatives.cpp (hunk 1 above)
Comment (DaniGP17) [reply code/components/extra-natives-five/src/WeatherNatives.cpp]: Sorry, I remembered that I had crashed on 3095 but it seems I was wrong, I just changed the version to 3258

**Comment by DaniGP17** [comment]
This should be closed because of this commit https://github.com/citizenfx/fivem/commit/0df3d8841ff7c466686f9798454b703cc6987d05?

**Comment by Mathu-lmn** [comment]
Yes, this issue was fixed so this can be safely closed !

**Comment by CoreBitX** [comment]
> Yes, this issue was fixed so this can be safely closed !

not fix!!

**Comment by DaniGP17** [comment]
> not fix!!

Please stop going through several issues saying the same thing without giving further explanation, this was closed because cfx already fixed it with the addition of new weather natives.

**Comment by CoreBitX** [comment]
> > not fix!!
> 
> Please stop going through several issues saying the same thing without giving further explanation, this was closed because cfx already fixed it with the addition of new weather natives.

Are you acting stupid or are you just pretending? Nothing has been fixed and it's not my fault. I've also used the new native and yet the game crashes, just like with 20 other players on my server. We also have 51 players together, it's all a broken game. I don't think so, stupid shitty child!

**Comment by DaniGP17** [comment]
> Are you acting stupid or are you just pretending? Nothing has been fixed and it's not my fault. I've also used the new native and yet the game crashes, just like with 20 other players on my server. We also have 51 players together, it's all a broken game. I don't think so, stupid shitty child!

I don't think insulting is the way to solve a problem. I'm trying to help you, but you still don't give any relevant information to help me find that supposed problem

**Pull request closed without merging**